jdaviz/app.py

```diff
@@ -1299,7 +1299,11 @@ def _update_selected_data_items(self, viewer_id, selected_items):
 
         # Include any selected data in the viewer
         for data_id, visibility in selected_items.items():
-            label = self._get_data_item_by_id(data_id)['name']
+            data_item = self._get_data_item_by_id(data_id)
+            if data_item is None:
+                label = None
```

Comment:
what scenario triggers this case?  I'm just trying to think if there's a way we can prevent it entirely, or if not if the warning that it will raise is warranted or will be to noisy.

Reply:
For this one, I cannot reproduce it, so I just looked at the traceback at https://github.com/spacetelescope/jdaviz/issues/689#issuecomment-1245931466 and did my best guess.

Reply:
ok, at least its not a common scenario then.  I don't think the extra logic to prevent an error is a problem... as long as it doesn't result in a lot of warnings for the user.  But I guess we'll at least have to reproduce (or find someone who can) before closing that issue.  I'll try to do some testing before approving, but otherwise the code changes themselves look good to me!  Thanks!